lib/ansible/cli/doc.py

```diff
@@ -1197,3 +1195,120 @@ def get_man_text(doc, collection_name='', plugin_type=''):
+        else:
+            desc = DocCLI.tty_ify(" ".join(opt['description']))
+
+        required = opt.get('required', False)
+        if not isinstance(required, bool):
+            raise("Incorrect value for 'Required', a boolean is needed.: %s" % required)
+
+        o = '%s:' % o
+        if module:
+            if required:
+                desc = "(required) %s" % desc
+            text.append("      %-20s   # %s" % (o, textwrap.fill(desc, limit, subsequent_indent=subdent)))
+        else:
+            if required:
+                default = '(required)'
+            else:
+                default = opt.get('default', 'None')
+
+            text.append("%s %-9s # %s" % (o, default, textwrap.fill(desc, limit, subsequent_indent=subdent, max_lines=3)))
+
+
+def _do_ini_snippet(text, doc):
+
+    #text.append("# %s:" % doc.get('plugin', doc.get('name')).upper())
+
+    subdent = "# "
+    limit = display.columns - 20
+
+    sections = {}
+    for o in sorted(doc['options'].keys()):
+        opt = doc['options'][o]
+        if isinstance(opt['description'], string_types):
+            desc = DocCLI.tty_ify(opt['description'])
+        else:
+            desc = DocCLI.tty_ify(" ".join(opt['description']))
+
+        required = opt.get('required', False)
+        if not isinstance(required, bool):
+            raise("Incorrect value for 'Required', a boolean is needed.: %s" % required)
+
+        if 'ini' in opt:
+            entry = opt['ini'][-1]
+            if entry['section'] not in sections:
+                sections[entry['section']] = []
+
+            if required:
+                default = '  # <REQUIRED>'
+            else:
+                default = opt.get('default', '')
+
+            key = textwrap.fill('# ' + desc, limit, max_lines=3, subsequent_indent=subdent) + '\n' + '%s=%s' % (entry['key'], default)
+            sections[entry['section']].append(key)
+
+    if sections:
+        for section in sections.keys():
+            text.append('[%s]' % section)
+            for key in sections[section]:
+                text.append(key)
+                text.append('')
+            text.append('')
+    else:
+        text.append('# no ini configuration options available')
+
+
+def _do_lookup_snippet(text, doc):
+
+    snippet = "lookup('%s', " % doc.get('plugin', doc.get('name'))
+    comment = ''
+    for o in sorted(doc['options'].keys()):
+
+        opt = doc['options'][o]
+        if o in ('_terms', '_raw', '_list'):
+            # these are 'list of arguments'
+            snippet += '< %s >' % (o)
+            comment = '# %s: %s' % (o, opt.get('description', ''))
+            continue
+
+        required = opt.get('required', False)
+        if not isinstance(required, bool):
+            raise("Incorrect value for 'Required', a boolean is needed.: %s" % required)
+
+        if required:
+            default = '<REQUIRED>'
+        else:
+            default = opt.get('default', 'None')
```

Comment:
`'None'` or `None` here?

I'd argue if it isn't required, and there is no default, then not displaying it at all is the better option.

Reply:
using string to show None, which is the default when no default is specified